### ISSUE TITLE
Discussion: Semantics of 'every n days' for recurring actions

### ISSUE THREAD

**Issue by C-Otto** [TracksApp/tracks#1845]
When creating a new recurring action to repeat itself every, say, 3 days, this works perfectly if the corresponding actions are completed on the intended date.

For example, I could complete an action on May 1st, then the next shows up with a show_from/due_date of May 4th. When I now complete this action on May 4th, this continues with a new action for May 7th. So far, so good.

However, as soon as I don't complete such an action on the intended date, tracks interpretes the 'every 3 days' setting as follows. The date of the new action is computed based on the initial show_from/due_date (which is in the past!), which could mean that the new action pops up directly after you completed the previous one.

This is not much of a difference when you have actions like 'do taxes once a year', where the exact date doesn't really matter. But for example when it comes to sports, intense workouts demand a day off between repetitions. With tracks, this is not possible - assuming one fails to do each task on the day it pops up.

Do you have any opinion on that? Are there advantages of the current behaviour, when compared to a variant where the date for the new action is derived based on the completion date of the previous action?

**Comment by C-Otto** [comment]
Using the option 'set the date the action should be shown (do not set a due date)' causes a different behavior. This helps in my case, but the (intended) meaning of this is not completely transparent.

**Comment by dnrce** [comment]
Does the "set the date the action should be shown" option completely solve this for you? Is it just a case of adjusting the wording to be more explanatory?

**Comment by C-Otto** [comment]
Yes, however it is not a matter of wording, but of a proper (lengthy) description. As far as I understand the code, a fixed due date is meant to be 100% rock solid. This also gets clear when you try to defer an action with a due date in the past, which is not possibly with the "Defer N days" shortcut. So if you think of due dates as something fixed to the specified date, the current behaviour is OK.

The documentation should make clear that, here and maybe in some other places, that a set due date is *not* meant to mean "should be done" but "must be done". Right now I don't know if GTD is clear about this (does it even mention due dates?), so maybe the documentation would be a bit redundant.

Using the other option, show from, is a great fix for recurring actions that don't have a fixed due date. In the past I often used a due date to indicate some sense of urgency and also let the action stand out from the others that can continue to sit there without being touched for a while. Using the show from solution helps with the issue I mentioned in this bug report, but I clearly need to be more strict about actually completing my actions.

tl;dr: The intention of strict due dates must be (made) clear, show from option helps with the specific issue I reported.

**Comment by ZeiP** [comment]
I think the logic of using the original due date is valid, but there are situations where both would make sense, as @C-Otto presented. I'm quite open to adding an option for this, if someone has time to implement it – it's not on the top of my to-do-list, however.
